### PR TITLE
QTSP-153: add domain and additional data to the logger

### PR DESCRIPTION
**Pull request by matsumonkie**

### CODE REVIEW

File: src/Effectful/Log.hs [hunk 1]
```diff
@@ -35,14 +36,18 @@ runLog
...
   -- ^ The maximum log level allowed to be logged.
+  -> [Text]
+  -- ^ The current application domain.
+  -> [Pair]
+  -- ^ Additional data to be merged with the log message's data
   -> Eff (Log : es) a
   -- ^ The computation to run.
   -> Eff es a
-runLog component logger maxLogLevel = evalStaticRep $ Log LoggerEnv
+runLog component logger maxLogLevel domain additionalData = evalStaticRep $ Log LoggerEnv
```
Comment (kawu) [inline src/Effectful/Log.hs]: Alternatively we could keep `runLog` as it is and provide an alternative interpreter `runLogWithEnv` (or some such name) that would take `LoggerEvn` on input.  The meaning of the arguments would be then clearer at the call site.  What do you think?

Comment (matsumonkie) [reply src/Effectful/Log.hs]: yup, it's better this way :+1:

File: log-effectful.cabal [hunk 2]
```diff
@@ -58,6 +58,7 @@ library
                    , log-base        >=0.12.0.0
                    , text
                    , time
+                   , aeson
```
Comment (mmhat) [inline log-effectful.cabal]: Why is that needed?

File: src/Effectful/Log.hs [hunk 3]
```diff
@@ -46,6 +47,15 @@ runLog component logger maxLogLevel = evalStaticRep $ Log LoggerEnv
   , leMaxLogLevel = maxLogLevel
   }
 
+-- | Run the 'Log' effect with full logging options
+runLogWithEnv
```
Comment (mmhat) [inline src/Effectful/Log.hs:51]: Maybe define `runLog` in terms of `runLogWithEnv`? I.e. `runLog component logger maxLogLevel = runLogWithEnv LoggerEnv ...`.

File: tests/Main.hs [hunk 4]
```diff
@@ -4,11 +4,20 @@ module Main where
...
   withStdOutLogger $ \logger -> do
     runLog "main" logger defaultLogLevel app
+  withStdOutLogger $ \logger -> do
+    let logEnv = LoggerEnv  { leLogger = logger
+                            , leComponent = "main"
+                            , leDomain = []
+                            , leData = [("testKey", Aeson.String "testValue") ]
+                            , leMaxLogLevel = defaultLogLevel
+                            }
+    runLogWithEnv logEnv app
```
Comment (mmhat) [inline tests/Main.hs]: I think changing the test is not necessary. Especially if `runLog` is defined in terms of `runLogWithEnv`; See previous comment.

**Review by mmhat — CHANGES_REQUESTED** [review]

File: log-effectful.cabal (hunk 2 above)
Comment (matsumonkie) [reply log-effectful.cabal]: it's not :+1:

File: src/Effectful/Log.hs (hunk 3 above)
Comment (matsumonkie) [reply src/Effectful/Log.hs:51]: :+1:

File: tests/Main.hs (hunk 4 above)
Comment (matsumonkie) [reply tests/Main.hs]: :+1:

**Review by kawu — APPROVED** [review]
Looks good to me, thanks!

File: src/Effectful/Log.hs [hunk 5]
```diff
@@ -38,14 +39,23 @@ runLog
...
-runLog component logger maxLogLevel = evalStaticRep $ Log LoggerEnv
+runLog component logger maxLogLevel = runLogWithEnv $ LoggerEnv
   { leLogger = logger
   , leComponent = component
   , leDomain = []
   , leData = []
   , leMaxLogLevel = maxLogLevel
   }
 
+-- | Run the 'Log' effect with full logging options
```
Comment (arybczak) [inline src/Effectful/Log.hs]:
```suggestion
-- | Run the 'Log' effect with full logging options.
```

**Comment by arybczak** [comment]
Please wait a bit with merging the PR, I'm not sure you need this (see question in jira).

**Review by mmhat — APPROVED** [review]
LGTM

@arybczak Ok, I'll leave the decision whether to merge this or not to you.

**Comment by matsumonkie** [comment]
Closing in favor of the already available `localData`.
